aws_lambda_builders/workflows/nodejs_npm/npm.py

```diff
@@ -0,0 +1,53 @@
+"""
+Wrapper around calling npm through a subprocess.
+"""
+
+import logging
+
+from .utils import OSUtils
+
+LOG = logging.getLogger(__name__)
+
+
+class NpmError(Exception):
+    pass
+
+
+class NpmExecutionError(NpmError):
+    def __init__(self, err):
+        super(NpmExecutionError, self).__init__(
+            'NPM failed: %s' % err)
+
+
+class SubprocessNpm(object):
+
+    def __init__(self, osutils=None, npm_exe=None):
+        if osutils is None:
+            osutils = OSUtils()
+        self.osutils = osutils
+
+        if npm_exe is None:
+            npm_exe = 'npm'
+
+        self.npm_exe = npm_exe
+
+    def main(self, args, cwd=None):
+
+        if not isinstance(args, list):
+            raise NpmExecutionError('args must be a list')
```

Comment:
So I would raise `ValueError` instead here. Users dont' call this method directly, so they don't need to know about the error when the caller did something wrong. `ValueError` is not caught any where and it would crash the process. 

I see `NpmExecutionError` as equivalent of HTTP 4xx (user error) and `ValueError` as equivalent of HTTP 5xx (internal error)

Reply:
resolved in https://github.com/gojko/aws-lambda-builders/commit/3e945b281e63e82d83e0fe3fe8f71b9c50f978aa